pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -278,7 +278,9 @@ private static PinotQuery compileCalciteSqlToPinotQuery(String sql) {
           pinotQuery.setOffset(Integer.valueOf(((SqlNumericLiteral) selectSqlNode.getOffset()).toValue()));
         }
         DataSource dataSource = new DataSource();
-        dataSource.setTableName(selectSqlNode.getFrom().toString());
+        if (selectSqlNode.getFrom() != null) {
```

Comment:
Can this be null? We shouldn't allow query without table name right?
I would suggest always putting a dummy table name in the test so that production code won't accidentally miss the table name

Reply:
From sql standard, it's ok if it's null, the query should not fail if it contains no literals.

Reply:
I don't quite follow. Why should we allow queries without the data source (i.e. table name)?
How are we going to execute query `SELECT *` without `FROM`?

Reply:
`SELECT *` is not valid as * is parsed as an identifier.

if you do `SELECT '*'` then it's a valid as you are querying for a string literal `*`

https://github.com/apache/incubator-pinot/pull/5513/files#diff-b1b494630a336fd5454600f3bcd0973dR52

Reply:
I see. So for literal only query, table name does not matter